File: algorithms/implementation/divisible-sum-pairs/divisible-sum-pairs-python/tests/test_divisibleSumPairs.py
from divisibleSumPairs import divisibleSumPairs

class TesteDivisibleSumPairs:
    
    def testeCaso1(self):
        
        n = 6
        k = 3
        ar = [1, 3, 2, 6, 1, 2]
        
        esperado = 5
        
        resultado = divisibleSumPairs(n, k, ar)
        
        assert resultado == esperado
        
        
    def testeCaso2(self):
        
        n = 2
        k = 2
        ar = [8, 10]
        
        esperado = 1
        
        resultado = divisibleSumPairs(n, k, ar)
        
        assert resultado == esperado
        
    def testeCaso3(self):
        
        n = 5
        k = 2
        ar = [5, 9, 10, 7, 4]
        
        esperado = 4
        
        resultado = divisibleSumPairs(n, k, ar)
        
        assert resultado == esperado
        
    def testeCaso4(self):
        
        n = 5
        k = 3
        ar = [2, 8, 6, 8, 4]
        
        esperado = 3
        
        resultado = divisibleSumPairs(n, k, ar)
        
        assert resultado == esperado
        
    def testeCaso5(self):
        
        n = 10
        k = 3
        ar = [29, 97, 52, 86, 27, 89, 77, 19, 99, 96]
        
        esperado = 15
        
        resultado = divisibleSumPairs(n, k, ar)
        
        assert resultado == esperado

File: algorithms/implementation/divisible-sum-pairs/divisible-sum-pairs-javascript/js/test_divisibleSumPairs.js
import { divisibleSumPairs } from "../js/divisibleSumPairs.js";
import assert from 'assert';

describe('divisibleSumPairs', function() {

    it('O número de pares é 5', function() {

        const n = 6;
        const k = 3;
        const ar = [1, 3, 2, 6, 1, 2]

        const esperado = 5;

        const resultado = divisibleSumPairs(n, k, ar)

        assert.strictEqual(resultado, esperado);
    });
});

describe('divisibleSumPairs', function() {

    it('O número de pares é 1', function() {

        const n = 2;
        const k = 2;
        const ar = [8, 10]

        const esperado = 1;

        const resultado = divisibleSumPairs(n, k, ar)

        assert.strictEqual(resultado, esperado);
    });
});

describe('divisibleSumPairs', function() {

    it('O número de pares é 4', function() {

        const n = 5;
        const k = 2;
        const ar = [5, 9, 10, 7, 4]

        const esperado = 4;

        const resultado = divisibleSumPairs(n, k, ar)

        assert.strictEqual(resultado, esperado);
    });
});


describe('divisibleSumPairs', function() {

    it('O número de pares é 3', function() {

        const n = 5;
        const k = 3;
        const ar = [2, 8, 6, 8, 4]

        const esperado = 3;

        const resultado = divisibleSumPairs(n, k, ar)

        assert.strictEqual(resultado, esperado);
    });
});

describe('divisibleSumPairs', function() {

    it('O número de pares é 15', function() {

        const n = 10;
        const k = 3;
        const ar = [29, 97, 52, 86, 27, 89, 77, 19, 99, 96]

        const esperado = 15;

        const resultado = divisibleSumPairs(n, k, ar)

        assert.strictEqual(resultado, esperado);
    });
});